### ISSUE TITLE
Cannot inspect parameter attributes in API Explorer with Minimal Actions

### ISSUE THREAD

**Issue by martincostello** [dotnet/aspnetcore#36438]
### Describe the bug

Attributes of parameters of Minimal Actions cannot be inspected in API Explorer as the `ParameterDescriptor` property of the `ApiParameterDescription` for the parameter is always `null`. Further, if it were not null, the type of the value of the property would need to implement `IParameterInfoParameterDescriptor` so that the relevant `ParameterInfo` value could be accessible for inspection.

https://github.com/dotnet/aspnetcore/blob/e6a5af6633f08890afbd99a6ca7687c623454768/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L166-L174

### To Reproduce

The following unit test demonstrates the issue, it fails due to `nameParam.ParameterDescription` being null.

```csharp
[Fact]
public void TestParameterAttributesCanBeInspected()
{
    var apiDescription = GetApiDescription(([Description("The name.")] string name) => { });
    Assert.Equal(1, apiDescription.ParameterDescriptions.Count);

    var nameParam = apiDescription.ParameterDescriptions[0];
    Assert.Equal(typeof(string), nameParam.Type);
    Assert.Equal(typeof(string), nameParam.ModelMetadata.ModelType);
    Assert.Equal(BindingSource.Query, nameParam.Source);
    Assert.False(nameParam.IsRequired);

    Assert.NotNull(nameParam.ParameterDescriptor); // Assertion fails here

    Assert.Equal("name", nameParam.ParameterDescriptor.Name);
    Assert.Equal(typeof(string), nameParam.ParameterDescriptor.ParameterType);

    var descriptor = Assert.IsAssignableFrom<IParameterInfoParameterDescriptor>(nameParam.ParameterDescriptor);

    Assert.NotNull(descriptor.ParameterInfo);

    var description = Assert.Single(descriptor.ParameterInfo.GetCustomAttributes<DescriptionAttribute>());

    Assert.NotNull(description);
    Assert.Equal("The name.", description.Description);
}
```

### Further technical details

Tested as of commit `b33b4d4453d93be12a847a5e0c222d86edd706a0`.

**Comment by martincostello** [comment]
Suggested fix in #36439.

**Comment by martincostello** [comment]
Related issue: even with the `ParameterInfo` exposed, Swashbuckle.AspNetCore does not discover annotations on the parameters by default due to an assumption that the concrete type of the descriptor is `ControllerParameterDescriptor`.

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/2224
